### PR TITLE
INTLY-9233

### PR DESCRIPTION
**Pull request by MStokluska**
# Description
Refactor of https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/marketplace/configmap_catalog_source_reconciler.go#L60 - addressing ticket https://issues.redhat.com/browse/INTLY-9233.
The function was refactored so that it uses `CreateOrUpdate` function.

# Verification

On a fresh cluster run:
- `make cluster/prepare/local`
- `make code/run`
RHMI Operator should successfully install with all the config maps created. Once installed, change some value in the manifests folder to test update functionality. For example:
- On your cluster open config map YAML for cloud resources 
- On your machine navigate to https://github.com/integr8ly/integreatly-operator/blob/master/manifests/integreatly-cloud-resources/0.18.0/cloud-resources.v0.18.0.clusterserviceversion.yaml#L116 and set the media type to image. 
- Once the cloud resources operator is reconciled you should see the config map update happening. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [x] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/resources/marketplace/configmap_catalog_source_reconciler.go [hunk 1]
```diff
@@ -68,26 +66,23 @@ func (r *ConfigMapCatalogSourceReconciler) reconcileRegistryConfigMap(ctx contex
...
-			}
-
-			logrus.Infof("Updated config map %s in namspace %s", configMapName, r.Namespace)
-		}
+	case controllerutil.OperationResultUpdated:
+		logrus.Infof("Updated config map %s in namspace %s", configMapName, r.Namespace)
+	case controllerutil.OperationResultNone:
+		break
+	default:
+		return configMapName, fmt.Errorf("Unknown controllerutil.OperationResult '%v'", or)
```
Comment (sergioifg94) [inline pkg/resources/marketplace/configmap_catalog_source_reconciler.go:85]: I would absent from returning an error in the `default`. It's very unlikely to reach this statement, but as long as the value of `err` is `nil` we shouldn't fail, maybe just log it. Anyways, it's very **very** unlikely to that this is going to happen so it's not a big deal.

```suggestion
	default:
		logrus.Infof("Unknown controllerutil.OperationResult '%v'", or)
```

Comment (MStokluska) [reply pkg/resources/marketplace/configmap_catalog_source_reconciler.go]: Thanks Sergio! That is a very good point! I've applied your suggestion.

**Comment by MStokluska** [comment]
Closed as the Jira got closed with "won't do"

**Pull request closed without merging**